ymmsl/checkpoint.py

```diff
@@ -0,0 +1,77 @@
+"""Definitions for describing checkpoints."""
+
+from typing import List, Optional, Union
+
+import yatiml
+
+
+class CheckpointRange:
+    def __init__(self,
```

Comment:
I'm missing the docstrings here. Without those, there's no API documentation, and without that, the users won't know what it does!